client/matching/client.go

```diff
@@ -21,46 +21,48 @@
 package matching
 
 import (
+	"context"
 	"sync"
 	"time"
 
-	"golang.org/x/net/context"
-
 	m "github.com/uber/cadence/.gen/go/matching"
+	"github.com/uber/cadence/.gen/go/matching/matchingserviceclient"
 	workflow "github.com/uber/cadence/.gen/go/shared"
 	"github.com/uber/cadence/common"
 	"github.com/uber/cadence/common/membership"
-	tchannel "github.com/uber/tchannel-go"
-	"github.com/uber/tchannel-go/thrift"
+	"go.uber.org/yarpc"
 )
 
 var _ Client = (*clientImpl)(nil)
 
 type clientImpl struct {
-	connection      *tchannel.Channel
-	resolver        membership.ServiceResolver
-	thriftCacheLock sync.RWMutex
-	thriftCache     map[string]m.TChanMatchingService
+	resolver          membership.ServiceResolver
+	thriftCacheLock   sync.RWMutex
+	thriftCache       map[string]matchingserviceclient.Interface
```

Comment:
do things ever get evicted from this cache ? If not, we will leak file descriptors everytime there is a host reboot (since we run on top of mesos / aurora) - let's create a separate issue for tackling this

Reply:
Oh..yes, i have a task filed to fix this right after this CR. the way to release is to call stop on dispatcher but i don't want to track dispatchers in this piece of code, but i have to, until i experiment with shard key where multiple outbound connections on single dispatcher. short term i get the connections are leaked untill reboot for next change.